### PR TITLE
plugins: Add obs-websocket as a submodule

### PR DESCRIPTION
**Pull request by tt2468**
### Description
Adds obs-websocket as a submodule, to be included in future releases of OBS Studio by default.

### Motivation and Context
This has been a long-time goal of the project. After over a year of work, obs-websocket version 5.0.0 is finally close to the finish line, so it's roughly time to go ahead and include it as a submodule.

### How Has This Been Tested?
Builds on Ubuntu 20.04, haven't tested elsewhere

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by PatTheMav** [comment]
You didn't simplify the CMakeLists over on obs-websocket yet, right? Because as-is I can see some stuff interfering. There should be a way to cordon off the parts required for standalone-building though.

File: CI/check-format.sh [hunk 1]
```diff
@@ -48,7 +48,8 @@ find . -type d \( \
     -path ./plugins/enc-amf -o \
     -path ./plugins/mac-syphon/syphon-framework -o \
     -path ./plugins/obs-outputs/ftl-sdk -o \
-    -path ./plugins/obs-vst \
+    -path ./plugins/obs-vst -o \
```
Comment (PatTheMav) [inline CI/check-format.sh:51]: `-path` takes a pattern, so I _think_ specifying `./plugins/obs-vst*` should match all paths that start with the pattern (including subdirs).

**Comment by PatTheMav** [comment]
Also `QTRandomGenerator` is only available with Qt 5.10, Ubuntu 18 delivers v5.9.5.

@RytoEX maybe time to drop ubuntu-18 from CI, given that we require CMake 3.16 anyway?

**Comment by tt2468** [comment]
> You didn't simplify the CMakeLists over on obs-websocket yet, right? Because as-is I can see some stuff interfering. There should be a way to cordon off the parts required for standalone-building though.

I did already go and merge the appropriate commits to our `master` branch. Before I did that though, I created a `release/5.0.0` branch which can be used to release standalone binaries for <28.0.0

**Comment by mbaldessari** [comment]
FWIW I tested this PR with obs-studio from master on Fedora 36 and the websocket works correctly (tested with a couple of simpleobsws based scripts)

**Pull request merged**